### ISSUE TITLE
Feature Request: Add Torch/Flashlight control to YOLOViewController

### ISSUE THREAD

**Issue by devstygar** [ultralytics/ultralytics#24643]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### The Problem
Currently, it is impossible to turn on the device's flash using third-party packages (like `torch_light` or `camera`) while `YOLOView` is active. Because `YOLOView` holds an exclusive lock on the camera hardware, the OS (both Android and iOS) blocks any external attempts to access the camera module's LED. 

Also, since the current `YOLOViewController` (v0.2.0) doesn't expose `pause()` or `resume()` methods for the camera stream, I cannot even implement a temporary workaround to free the camera, turn on the flash, and resume.

### Proposed Solution
It would be incredibly helpful to have native flashlight control exposed directly through the `YOLOViewController`. 

For example, adding methods like:
- `yoloController.toggleTorch()`
- `yoloController.enableTorch(bool enable)`
- `yoloController.setFlashMode(FlashMode)`

Since the package already manages the native camera lifecycle, having this feature built-in is the only reliable way to use the flash during real-time detection without hitting OS hardware locks.

Thank you for the amazing package!

### Use Case
I am developing an application that uses `YOLOView` for real-time object detection. Often, the users need to scan items in low-light environments (e.g., dark warehouses or at night), which requires turning on the device's flashlight/torch.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lucas-stygar, thank you for your feature request for Ultralytics 🚀 and for the clear description of the proposed `YOLOViewController` torch/flashlight controls. This is an automated response to help get things moving quickly, and an Ultralytics engineer will also assist you soon 🤖✨

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

For this feature request, it would be helpful if you could also share any additional implementation context, such as:
- target platform details for Android and iOS
- the package/version where `YOLOViewController` is being used
- any relevant logs or snippets showing current camera/torch behavior
- whether you need simple on/off torch support or broader flash mode control as well 💡

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by devstygar** [comment]
Hello! Here is the additional context requested:

* **Target platform details:** Both Android and iOS (built with Flutter).
* **Package/version:** `ultralytics_yolo` version `0.2.0`.
* **Current behavior:** The `YOLOView` widget locks the camera hardware. Any attempt to use a standard Flutter package like `torch_light` throws an OS-level exception or is silently ignored because the camera stream holds exclusive access to the module.
* **Scope of feature:** A simple ON/OFF torch support (e.g., `enableTorch(bool)`) would completely solve the problem for scanning tags in low-light environments. Broader flash mode control would be a great bonus, but a simple toggle is all that is strictly necessary.

**Comment by glenn-jocher** [comment]
Thanks for the added context — a native `enableTorch(bool)` on `YOLOViewController` looks like a reasonable enhancement for low-light Flutter use, so we’ll keep this request open for review; if you’d like to help move it forward, a small, focused PR adding Android/iOS torch support with minimal docs/tests would be welcome.

**Comment by Sunny11092003** [comment]
Hi @glenn-jocher,

I'd like to work on this feature and submit a PR for torch/flashlight support in YOLOViewController for Android and iOS.

I'll review the current implementation and start working on a minimal solution with documentation.

Thank you!

**Comment by glenn-jocher** [comment]
Thanks — please go ahead and open a small, focused PR for `YOLOViewController` torch support on Android/iOS, keep it limited to this feature without unrelated config changes, and include minimal docs/tests; the [contributing guide](https://docs.ultralytics.com/help/contributing) covers the PR checklist and CLA step.

**Comment by onuralpszr** [comment]
@lucas-stygar @Sunny11092003 for IOS issues please open issue in here https://github.com/ultralytics/yolo-ios-app
For android/flutter please open https://github.com/ultralytics/yolo-flutter-app here thank you people :)
